### DISCUSSION TITLE
filter from value from nested realtionship

### DISCUSSION THREAD

**Discussion by sumeet030990** [prisma/prisma#14658] [Q&A]
I want to search for orders where the product has the item_id entered by the user.
for eg: item_id: 123
 
My schema
```
model orders {
  id                 Int            @id @default(autoincrement())
  order_details      details[]
}
```
```
//order details
model details {
  id           Int       @id @default(autoincrement())
  orders     orders?   @relation(fields: [order_id], references: [id])
  order_id   Int?
  products   products? @relation(fields: [product_id], references: [id])
  product_id Int?
}
```
```
model products {
  id                 Int               @id @default(autoincrement())
  item_id            String?           @unique @db.VarChar(30)
  order_details      details[]
}
```

I am not sure exaclty what to write in where block 
```
const result = await prisma.orders.findMany({
    where: {
      order_details: {
        some: { // here its saying some, every, none are possible values 
          products: {
            is: { // is and isnot are the possible values
              item_id,
            },
          },
        },
      },
    }
  });
```

error
```
Unknown arg `is` in where.order_details.some.products.is for type productsWhereInput. Did you mean `id`? Available args:\ntype productsWhereInput {\n  AND?: productsWhereInput | List<productsWhereInput>\n  OR?: List<productsWhereInput>\n  NOT?: productsWhereInput | List<productsWhereInput>\n  id?: IntFilter | Int\n  manufacturer?: StringNullableFilter | String | Null\n  description?: StringNullableFilter | String | Null\n  net_weight?: StringNullableFilter | String | Null\n  case_count?: IntNullableFilter | Int | Null\n  case_cost?: IntNullableFilter | Int | Null\n  unit_cost?: IntNullableFilter | Int | Null\n  msrp?: StringNullableFilter | String | Null\n  upc_retail?: StringNullableFilter | String | Null\n  changed?: StringNullableFilter | String | Null\n  gp_id?: StringNullableFilter | String | Null\n  types?: JsonNullableFilter\n  enabled?: BoolNullableFilter | Boolean | Null\n  is_new?: BoolNullableFilter | Boolean | Null\n  is_holiday?: BoolNullableFilter | Boolean | Null\n  disabled_date?: DateTimeNullableFilter | DateTime | Null\n  item_id?: StringNullableFilter | String | Null\n  image?: StringNullableFilter | String | Null\n  length?: IntNullableFilter | Int | Null\n  width?: IntNullableFilter | Int | Null\n  depth?: IntNullableFilter | Int | Null\n  holiday_start_date?: DateTimeNullableFilter | DateTime | Null\n  holiday_end_date?: DateTimeNullableFilter | DateTime | Null\n  products_stores?: Products_storesListRelationFilter\n  order_details?: DetailsListRelationFilter\n}\n\n"
```

**Comment by nmfrankel** [comment]
Try replacing the following lines...
```js
    products: {
          is: { // is and isnot are the possible values
              item_id,
          },
    }
```
to 

```js
    products: item_id
```

**Reply by nmfrankel** [reply]
the `some: { ... }` object might also not be needed.

**Reply by vladi-stevanovic** [reply]
👋 Hello @sumeet030990 I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Reply by sumeet030990** [reply]
I resolved the issue using ILIKE in raw query but I was hoping it would be better if I could achieve this via ORM style instead of raw query
as because just for implementing the search feature I have to convert whole query and joints into raw query

**Reply by pantharshit00** [reply]
Hello @sumeet030990 

Sorry for late reply here. You can try using case insensitive filtering here which will use `ILIKE` under the hood: https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting#case-insensitive-filtering

Let me know if this works for you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
